src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ReleaseDateService.kt

```diff
@@ -41,48 +40,33 @@ class ReleaseDateService(
   }
 
   fun getEarliestReleaseDate(releaseDate: LocalDate): LocalDate {
-    return getEarliestDateBefore(maxNumberOfWorkingDaysAllowedForEarlyRelease, releaseDate, ::isEligibleForEarlyRelease)
+    return getEarliestDateBefore(maxNumberOfWorkingDaysAllowedForEarlyRelease, releaseDate)
   }
 
   fun isLateAllocationWarningRequired(releaseDate: LocalDate?): Boolean {
     if (releaseDate === null || releaseDate.isBefore(LocalDate.now(clock))) return false
     val warningThreshold = getEarliestDateBefore(
       maxNumberOfWorkingDaysToTriggerAllocationWarningEmail,
       releaseDate,
-      ::isBankHolidayOrWeekend,
     )
     return LocalDate.now(clock) >= warningThreshold
   }
 
-  /** Friday is not considered as weekend */
-  fun isBankHolidayOrWeekend(releaseDate: LocalDate?): Boolean {
-    val dayOfWeek = releaseDate?.dayOfWeek
-    if (dayOfWeek == DayOfWeek.SATURDAY || dayOfWeek == DayOfWeek.SUNDAY) {
-      return true
-    }
-    val listOfBankHolidays: List<LocalDate> = bankHolidayService.getBankHolidaysForEnglandAndWales()
-    return listOfBankHolidays.contains(releaseDate)
-  }
-
   /** Friday is also considered as weekend */
   fun isEligibleForEarlyRelease(releaseDate: LocalDate): Boolean {
-    val listOfBankHolidays: List<LocalDate> = bankHolidayService.getBankHolidaysForEnglandAndWales()
-    val dayOfWeek = releaseDate.dayOfWeek
-    if (dayOfWeek == DayOfWeek.FRIDAY || dayOfWeek == DayOfWeek.SATURDAY || dayOfWeek == DayOfWeek.SUNDAY) {
-      return true
-    }
-    return listOfBankHolidays.contains(releaseDate)
+    return workingDaysService.isNonWorkingDay(
+      releaseDate,
+      listOf(DayOfWeek.FRIDAY, DayOfWeek.SATURDAY, DayOfWeek.SUNDAY),
```

Comment:
Friday is a working day, could pull the Friday date check up to this level? and not parameterise the `isNonWorkingDay` fun?

Reply:
Makes sense!